addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -158,6 +192,58 @@ weSnippetEditor.SnippetsMenu.include({
         this._super(...arguments);
         this.$('.o_we_customize_theme_btn').toggleClass('active', tab === this.tabs.THEME);
     },
+    /**
+     * Returns the animated text element wrapping the selection if it exists.
+     *
+     * @private
+     * @param {Node} editable
+     * @return {Element|false}
+     */
+    _getAnimatedTextElement(editable) {
+        const animatedTextNode = getTraversedNodes(editable).find(n => n.parentElement.closest(".o_animated_text"));
+        return animatedTextNode ? animatedTextNode.parentElement.closest('.o_animated_text') : false;
+    },
+    /**
+     * @override
+     */
+    _addToolbar() {
+        this._super(...arguments);
+        this.$('#o_we_editor_toolbar_container > we-title > span').after($(`
+            <div class="btn fa fa-fw fa-2x o_we_highlight_animated_text d-none
+                ${$('body').hasClass('o_animated_text_highlighted') ? 'fa-eye text-success' : 'fa-eye-slash'}"
+                title="Highlight Animated Text"
+                aria-label="Highlight Animated Text"/>`)
+        );
+        this._toggleAnimatedTextButton();
+        this._toggleHighlightAnimatedTextButton();
+    },
+    /**
+     * Activates the button to animate text if the selection is in an
+     * animated text element or deactivates the button if not.
+     * 
+     * @private
+     */
+    _toggleAnimatedTextButton() {
+        const editable = this.options.wysiwyg.$editable[0];
```

Comment:
On L243, you use `this.getEditableArea()`, is this the same as this stuff? `getEditableArea` sound better.

Reply:
this.getEditableArea() => list of all the elements editable (".o_editable")
this.options.wysiwyg.$editable[0]; => "#wrapwrap"

Reply:
Ok but then you always want to look inside the individual editable element, no? Not the whole wrapwrap which contains many non editable stuff

Reply:
I'm not sure I understand what you want? All the editor functions used here take a node as a parameter. Would you like me to loop all the o_editable elements?

Reply:
What I want is what is correct, I'm actually asking you here :wink: From my point of view, it seems necessary to consider only editable elements and so yes, loop over them but I did not go as far in the code as you while developing so I may be wrong: in the case, I guess a comment explaining why this is not necessary is needed as not obvious from a reviewer point of view :+1: